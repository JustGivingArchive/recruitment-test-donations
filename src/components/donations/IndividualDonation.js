import React from 'react';

const IndividualDonation = ({amount, message, imageUrl}) => {
  return (
    <div className="panel">
      <div className="donorinfo">
      <img
        className="donorimage"
        alt="donor profile picture"
        src={imageUrl}></img>
      <p className="paneltextlarge">Â£{amount || '--'}</p>
      </div>
      <p className="paneltext donormessage">{message}</p>
    </div>
  );
};

export default IndividualDonation;
