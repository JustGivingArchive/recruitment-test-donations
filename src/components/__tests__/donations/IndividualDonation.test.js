import React from 'react';
import renderer from 'react-test-renderer';
import IndividualDonation from '../../donations/IndividualDonation.js';

describe('Test donation info componenet', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <IndividualDonation />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
