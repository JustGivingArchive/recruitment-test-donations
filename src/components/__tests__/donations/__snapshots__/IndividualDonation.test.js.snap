exports[`Test donation info componenet renders correctly 1`] = `
<div
  className="panel">
  <div
    className="donorinfo">
    <img
      alt="donor profile picture"
      className="donorimage"
      src={undefined} />
    <p
      className="paneltextlarge">
      Â£
      --
    </p>
  </div>
  <p
    className="paneltext donormessage" />
</div>
`;
